Standard error shading and svg cleanup

anotherjoshsmith opened the pull request:
I've added +/- 1 standard error of the mean shading (per @jyeatman AFQ [paper](https://doi.org/10.1371/journal.pone.0049790)). Still need to add a few things, but figured I'd get this PR rolling in case y'all want to check out the shading before the meeting tomorrow! 

Outstanding issues before merge:
- [x]  Append tract name text
- [x]  Append y axis label
- [ ] Adjust plot svg to avoid data drawn over x axis

anotherjoshsmith committed:
Calculate stderror of mean

anotherjoshsmith committed:
Add std error shading to ready function

anotherjoshsmith committed:
Update standard error with plot change

anotherjoshsmith commented:
Ah forgot a sqrt in there, sorry! On it now.

arokem commented:
This looks great. Unless I am missing something, it looks (from the code) like you are plotting the standard *deviation*, rather than the standard *error*. If I understand correctly, at least that's what `d3.deviation` will return, so unless you are transforming it somewhere else, that's what these shaded regions represent?

On the other hand -- I also think that's what we *want* to do, based on a brief conversation with @jyeatman about this last week (and he can confirm). The gist of it is that at least for comparisons of individuals to groups, we are interested in whether they differ by std, rather than stderr. 

If this is all true, we might just want to change the variable name to `std` instead of `stderr`.

jyeatman commented:
Ideally we could switch between std and stderr since they have different
applications. For figure 2 of the paper we would want stderr for the group
comparison. For figure 3 we would want std so we can compare individual
patients to the distribution of controls
On Mon, May 22, 2017 at 8:27 PM Ariel Rokem <notifications@github.com>
wrote:

> This looks great. Unless I am missing something, it looks (from the code)
> like you are plotting the standard *deviation*, rather than the standard
> *error*. If I understand correctly, at least that's what d3.deviation
> will return, so unless you are transforming it somewhere else, that's what
> these shaded regions represent?
>
> On the other hand -- I also think that's what we *want* to do, based on a
> brief conversation with @jyeatman <https://github.com/jyeatman> about
> this last week (and he can confirm). The gist of it is that at least for
> comparisons of individuals to groups, we are interested in whether they
> differ by std, rather than stderr.
>
> If this is all true, we might just want to change the variable name to std
> instead of stderr.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/yeatmanlab/AFQ-Browser/pull/145#issuecomment-303255906>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABsYyrjyGfENaGgpmGitGP3mWcLEiF2Dks5r8ie-gaJpZM4Ni943>
> .
>
-- 
Sent from a mobile device. Please excuse typos

anotherjoshsmith committed:
Std as default err with stderr option

anotherjoshsmith commented:
@arokem As we construct afqb.plots.tractMean in the ready() and updatePlots() functions, the return statements are internal to d3.mean() and d3.deviation(). For stddev, I divide the result of d3.deviation() by sqrt(n).

@arokem @jyeatman Most recent commit plots std as default error type, with stderr option in plots control box.

jyeatman commented:
@anotherjoshsmith this looks great. The aesthetics and control box is very nice. But the standard error looks too small.  Are we missing is it computed as std/sqrt(n)? Or is the sqrt missing? Or maybe n is the full n rather than the n of each group?

anotherjoshsmith commented:
@jyeatman sqrt is in there. Once you split, it uses the n in the first group, which may make SE to small for other groups.

jyeatman commented:
![std](https://cloud.githubusercontent.com/assets/1775818/26368848/d9b3d50e-3fc0-11e7-9433-dedb3a13282a.jpg)
![errorbar](https://cloud.githubusercontent.com/assets/1775818/26368852/dd3ad75e-3fc0-11e7-9092-d066ca5069b3.jpg)

jyeatman commented:
Upper two figures show FA values for two groups for the Left CST with standard error and standard deviation. For some reason the standard deviations look consistent with AFQ-Browser, but the standard errors look way smaller in AFQ-Browser

@arokem any ideas, or am I doing this wrong in matlab?

jyeatman commented:
sorry should have said that the above plots were made by doing the group split based on Age based on the data here

https://yeatmanlab.github.io/afqbrowser-demo/

jyeatman commented on afqbrowser/site/client/js/tract-details.js:
```diff
@@ -293,15 +320,23 @@ function ready(error, data) {
+						 return +d[afqb.controls.plotsControlBox.plotKey];}),
+					  stderr: d3.deviation(v, function (d) {
+						 return +d[afqb.controls.plotsControlBox.plotKey];
+					   })/Math.sqrt(afqb.plots.tractData[0].values.length),
```
@anotherjoshsmith @arokem Does the length of this have just the subjects in this group or is it a binary grouping variable where the length is as long as all the subjects? Based on appearance of the plots it looks like n is the full n not the group's n

anotherjoshsmith replied:
@jyeatman The ready function uses n of the whole data set because we generate the plots prior to any split. Line 487 in tract-details.js (in updatePlots() function) uses the size of the first split group instead.

anotherjoshsmith committed:
y axis and tract name labels

arokem commented:
@jyeatman : if I understand correctly, the error bars in this case should become about 6 times smaller (sqrt(76/2), where 76 is the total sample size). Is that correct? Is that we're getting in Matlab? We're checking that this is what happens on this side.

jyeatman commented:
yes, that is the case.
On Tue, May 23, 2017 at 4:49 PM Ariel Rokem <notifications@github.com>
wrote:

> @jyeatman <https://github.com/jyeatman> : if I understand correctly, the
> error bars in this case should become about 6 times smaller (sqrt(76/2),
> where 76 is the total sample size). Is that correct? Is that we're getting
> in Matlab? We're checking that this is what happens on this side.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/yeatmanlab/AFQ-Browser/pull/145#issuecomment-303527546>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABsYymC5Bp_saRCox0PVJlZRnm8jqEa6ks5r80YmgaJpZM4Ni943>
> .
>
-- 
Sent from a mobile device. Please excuse typos

arokem commented:
OK - we've found the cause. We're always using the length of the first group as 'n' for the sqrt(n) correction. In this case, the first group has 50 subjects (because we are dividing such that the number of unique values in each group is identical), and the other group has 27 subjects. But we're dividing by sqrt(50 for both of them. And that's why the error bars look really small here. This would not be the case for more even groups, but we should still figure out a way to look at each group separately for 'n'.

jyeatman commented:
Ah - that makes sense
On Tue, May 23, 2017 at 5:08 PM Ariel Rokem <notifications@github.com>
wrote:

> OK - we've found the cause. We're always using the length of the first
> group as 'n' for the sqrt(n) correction. In this case, the first group has
> 50 subjects (because we are dividing such that the number of unique values
> in each group is identical), and the other group has 27 subjects. But we're
> dividing by sqrt(50 for both of them. And that's why the error bars look
> really small here. This would not be the case for more even groups, but we
> should still figure out a way to look at each group separately for 'n'.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/yeatmanlab/AFQ-Browser/pull/145#issuecomment-303532439>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABsYyn-KE448NdsyDlXzDpUlajj5dYX1ks5r80p3gaJpZM4Ni943>
> .
>
-- 
Sent from a mobile device. Please excuse typos

jyeatman commented:
Just took a look at the updates and they look great. Once we fix the N then I think this is ready to merge

arokem committed:
Merge branch 'master' into updatePlots

arokem commented:
Same conflict as in #139 resolved.

anotherjoshsmith committed:
Fix stderr calculation

anotherjoshsmith committed:
Merge branch 'updatePlots' of https://github.com/anotherjoshsmith/AFQ-Browser into updatePlots

resolve merge errors due to changes in css file

anotherjoshsmith commented:
Stderr should be calculated with the correct number of subjects for each group now. @jyeatman let me know if you still see any issues when comparing to the matlab plots!

anotherjoshsmith commented:
Ah looks like there's a couple issues with the css for area. The original error plot area is not filled in and there's a thin gray border around all area paths. Will fix this with another commit!

jyeatman commented:
stderr still looks too small - when I plot  stderr it is thinner than the mean line

anotherjoshsmith commented:
As far as I can tell, the math is correct now. Could be an issue with visualization in one tool or the other. Here's a screenshot of CFMajor with group sizes 50 and 25. Shows thin stderr area, but not completely hidden by mean lines.
<img width="569" alt="screen shot 2017-07-10 at 7 39 14 am" src="https://user-images.githubusercontent.com/11064269/27998824-c712c30e-6547-11e7-921b-088afbc54e23.png">

I also printed out the number of subjects in each group as calculated by (std/stderr)^2 for splits on different columns. 75 total subjects.
<img width="248" alt="screen shot 2017-07-10 at 8 10 00 am" src="https://user-images.githubusercontent.com/11064269/27998825-c9ee78ac-6547-11e7-8cf2-f72a19dfa14e.png">

jyeatman commented:
Oh, I think you're right. Looks like my confusion was just because the std
err was light.


On Sun, Jul 9, 2017 at 5:20 PM, Josh Smith <notifications@github.com> wrote:

> As far as I can tell, the math is correct now. Could be an issue with
> visualization in one tool or the other. Here's a screenshot of CFMajor with
> group sizes 50 and 25. Shows thin stderr area, but not completely hidden by
> mean lines.
> [image: screen shot 2017-07-10 at 7 39 14 am]
> <https://user-images.githubusercontent.com/11064269/27998824-c712c30e-6547-11e7-921b-088afbc54e23.png>
>
> I also printed out the number of subjects in each group as calculated by
> (std/stderr)^2 for splits on different columns. 75 total subjects.
> [image: screen shot 2017-07-10 at 8 10 00 am]
> <https://user-images.githubusercontent.com/11064269/27998825-c9ee78ac-6547-11e7-8cf2-f72a19dfa14e.png>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/yeatmanlab/AFQ-Browser/pull/145#issuecomment-313975293>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABsYyvNxmxePctH57-XwFzFFm8fAjjZ0ks5sMW5HgaJpZM4Ni943>
> .
>

arokem commented:
So is this ready to be merged?

On Sun, Jul 9, 2017 at 5:44 PM, Jason D. Yeatman <notifications@github.com>
wrote:

> Oh, I think you're right. Looks like my confusion was just because the std
> err was light.
>
>
> On Sun, Jul 9, 2017 at 5:20 PM, Josh Smith <notifications@github.com>
> wrote:
>
> > As far as I can tell, the math is correct now. Could be an issue with
> > visualization in one tool or the other. Here's a screenshot of CFMajor
> with
> > group sizes 50 and 25. Shows thin stderr area, but not completely hidden
> by
> > mean lines.
> > [image: screen shot 2017-07-10 at 7 39 14 am]
> > <https://user-images.githubusercontent.com/11064269/27998824-c712c30e-
> 6547-11e7-921b-088afbc54e23.png>
> >
> > I also printed out the number of subjects in each group as calculated by
> > (std/stderr)^2 for splits on different columns. 75 total subjects.
> > [image: screen shot 2017-07-10 at 8 10 00 am]
> > <https://user-images.githubusercontent.com/11064269/27998825-c9ee78ac-
> 6547-11e7-8cf2-f72a19dfa14e.png>
> >
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/yeatmanlab/AFQ-Browser/pull/
> 145#issuecomment-313975293>,
> > or mute the thread
> > <https://github.com/notifications/unsubscribe-auth/ABsYyvNxmxePctH57-
> XwFzFFm8fAjjZ0ks5sMW5HgaJpZM4Ni943>
> > .
>
> >
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/yeatmanlab/AFQ-Browser/pull/145#issuecomment-313976718>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAHPNpb1bWWFf-QYYnBF9gGGxllvS1YOks5sMXPagaJpZM4Ni943>
> .
>

anotherjoshsmith commented:
@arokem Ready with the exception of some small .css issue that I haven't looked into yet. I'll work on a final commit first thing in the morning (China time)!

anotherjoshsmith committed:
Fix path.area style

anotherjoshsmith commented:
Removed the gray border and added fill for the initial error plot. Let me also see if I can quickly check the final box in the initial comment of this PR, I missed that yesterday.

anotherjoshsmith commented:
@arokem @jyeatman The unchecked box from the initial PR comment is about reorganizing our calls to different elements in the y plot. I will work on the outstanding issue on a different branch. 

I think it is best to merge this PR to add the error feature and plot labels to master.

jyeatman commented:
This looks great. Tested it and all good
